examples/src/croncat.rs

```diff
@@ -0,0 +1,204 @@
+// This example will go over the croncat contracts found
+// [here](https://github.com/CronCats/contracts/blob/cafd3caafb91b45abb6e811ce0fa2819980d6f96/manager/src/lib.rs)
+// This will demonstrate a more involved example for fast-forwarding the blockchain to a future
+// state/time. This is useful for testing anything that is time dependent such as scheduling
+// This is perfect to showcase cron.cat which will schedule calling into contract functions
+// at a set amount of time we supply.
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_sdk::json_types::U128;
+use near_units::{parse_gas, parse_near};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+use workspaces::network::Sandbox;
+use workspaces::prelude::*;
+use workspaces::{Account, AccountId, Contract, Worker};
+
+const MANAGER_CONTRACT: &[u8] = include_bytes!("../res/manager.wasm");
+const COUNTER_CONTRACT: &[u8] = include_bytes!("../res/counter.wasm");
+
+/// `AgentStatus` struct taken from [croncat repo](github.com/CronCats/contracts/) to
+/// deserialize into after we get the result of a transaction and converting over to
+/// this particular type.
+#[derive(BorshDeserialize, BorshSerialize, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(crate = "near_sdk::serde")]
+pub enum AgentStatus {
+    Active,
+    Pending,
+}
+
+/// `Agent` struct taken from [croncat repo](github.com/CronCats/contracts/) to deserialize
+/// into after we get the result of a transaction and converting over to this particular type.
+/// Helpful for understanding what our output is from a contract call. For a more in depth
+/// look at what an `Agent` is all about, refer to the [croncat docs](https://docs.cron.cat/docs/)
+/// to understand further, but for this example all we care about is that an Agent is something
+/// that can run scheduled tasks once it is time and collect rewards thereafter.
+#[derive(BorshDeserialize, BorshSerialize, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(crate = "near_sdk::serde")]
+pub struct Agent {
+    pub status: AgentStatus,
+    pub payable_account_id: AccountId,
+    pub balance: U128,
+    pub total_tasks_executed: U128,
+    pub last_missed_slot: u128,
+}
+
```

Comment:
prob shouldn't import and compile the SDK for just this, prob better to just include the util here. Guess it doesn't have to come with this, but avoiding linking to SDK for examples is probably best.

Something like:

```suggestion
    #[serde(with = "string")]
    pub balance: u128,
    #[serde(with = "string")]
    pub total_tasks_executed: u128,
    pub last_missed_slot: u128,
}

mod string {
    use std::fmt::Display;
    use std::str::FromStr;

    use serde::{de, Serializer, Deserialize, Deserializer};

    pub fn serialize<T, S>(value: &T, serializer: S) -> Result<S::Ok, S::Error>
        where T: Display,
              S: Serializer
    {
        serializer.collect_str(value)
    }

    pub fn deserialize<'de, T, D>(deserializer: D) -> Result<T, D::Error>
        where T: FromStr,
              T::Err: Display,
              D: Deserializer<'de>
    {
        String::deserialize(deserializer)?.parse().map_err(de::Error::custom)
    }
}
```

Also, a bit unclear why the last_missed_slot would be an integer when the others aren't. I'm sure there is a motivation, but it isn't clear to me

Reply:
> Also, a bit unclear why the last_missed_slot would be an integer when the others aren't. I'm sure there is a motivation, but it isn't clear to me

no idea, copied it directly from croncat. @TrevorJTClarke can probably explain here